src/blockchain/rpc.rs

```diff
@@ -197,132 +207,151 @@ impl Blockchain for RpcBlockchain {
         db: &mut D,
         _progress_update: P,
     ) -> Result<(), Error> {
-        let mut indexes = HashMap::new();
-        for keykind in &[KeychainKind::External, KeychainKind::Internal] {
-            indexes.insert(*keykind, db.get_last_index(*keykind)?.unwrap_or(0));
-        }
+        // internal cache to speed up conftime lookups
+        let mut txid_conftime_map = BTreeMap::new();
 
-        let mut known_txs: HashMap<_, _> = db
-            .iter_txs(true)?
-            .into_iter()
-            .map(|tx| (tx.txid, tx))
-            .collect();
-        let known_utxos: HashSet<_> = db.iter_utxos()?.into_iter().collect();
+        let mut request =
+            super::script_sync::start(db, self.stop_gap.unwrap_or(DEFAULT_STOP_GAP) as usize)?;
 
-        //TODO list_since_blocks would be more efficient
-        let current_utxo = self
-            .client
-            .list_unspent(Some(0), None, None, Some(true), None)?;
-        debug!("current_utxo len {}", current_utxo.len());
+        // Fetch txs from core in batches of 1000 txs
+        // Stop when there's no more
+        let mut listed_txs = vec![];
+        let mut fetch_count = 0;
+        loop {
+            let fetched_txs = self.client.list_transactions(
+                None,
+                Some((fetch_count + 1) * 1000),
+                Some(fetch_count * 1000),
+                Some(true),
+            )?;
+            listed_txs.extend(fetched_txs.clone());
+            if fetched_txs.len() < 1000 {
+                break;
+            }
```

Comment:
`fetched_txs.len()` should be saved in a var so that the clone is not needed

Reply:
Noted..